Organisations page

tinashechiraya opened the pull request:
[Screencast from 31-08-23 13:34:56.webm](https://github.com/kartoza/sawps/assets/70011086/2df8b381-5bae-4e8a-8ad3-27b7c8f544be)

Descriptions:
the switch organisation has been removed from the dropdown in the navigation from the header to this page
on this page user can see the list of organisations they have access to 
the user can choose the organisation they wish to set active 
clicking on an organisation that is currently active will navigate to the organisation users page

dimasciput commented:
@tinashechiraya selected organisatation need to be stored in database not in session anymore

tinashechiraya commented:
@dimasciput Yes I made a comment on that , I asked if I should proceed with making another ticket on that because changing the the active organisation from being stored in the session to the db will also affect other parts of the code so I will have to update them as well 

![2023-09-01_06-37](https://github.com/kartoza/sawps/assets/70011086/f77755d5-f7f6-4e84-9f9e-a9c7bf9105a8)

dimasciput commented:
@tinashechiraya thanks, can you add it here please

tinashechiraya commented:
@dimasciput Alright no problem

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 21 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 21 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 47 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 21 min

tinashechiraya commented:
@dimasciput I have now added a new field on the user profile that will store their current/active organisation .
I have also replaced all the places in the code that were using the current organisation from the session , it will now use the new utility function to fetch the current organisation from the user profile.
I have also update a few tests so they don't break

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 41 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

lgkgh commented:
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

dimasciput commented on django_project/frontend/api_views/map.py:
```diff
@@ -250,22 +253,26 @@ class FindPropertyByCoord(APIView):
         point = Point(float(lng), float(lat), srid=4326)
+        current_organisation_id = get_current_organisation_id(
+            request.user
+        ) or 0
```
Could you remove this? Just let it returns None.

dimasciput commented on django_project/stakeholder/test_view.py:
```diff
@@ -673,7 +693,7 @@ def test_notifications_page(self):
         request = self.factory.post(url, data)
         request.user = self.user
-        request.session = {CURRENT_ORGANISATION_ID_KEY: self.organisation.id}
+        # request.session = {CURRENT_ORGANISATION_ID_KEY: self.organisation.id}
```
Please just remove all commented codes.
